vamp-json/VampJson.h

```diff
@@ -1109,11 +1109,14 @@ class VampJson
 
     static bool
     successful(json11::Json j, std::string &err) {
-        if (!j["result"].is_object()) {
-            err = "result object expected for success";
+        if (j["result"].is_object()) {
+            return true;
+        } else if (j["error"].is_object()) {
+            return false;
+        } else {
```

Comment:
I think these checks should be the other way around. If we have an error, then there's a failure even if we also have a result. Or perhaps, if we have both an error and a result, then we should be setting err instead (as presumably this shouldn't really happen). Either way I don't think we should return true if the error field is present.